### DISCUSSION TITLE
Any plans to support the new JK PB series BMS

### DISCUSSION THREAD

**Discussion by kommando828** [ArminJo/JK-BMSToPylontechCAN#34] [General]
I have two of the new JK PB BMS's, it has Canbus included and I run it open loop with a pair of SMA Sunny Island's SI5048 in closed loop. It works with Pylontech, Deye, Goodwe and FSS (SMA Canbus for SI 6.0/8.0H-11 which is possibly very close to Pylontech Plus) Canbus set on the JK.

In the main it works well, however the inaccurate SOC looks to have been carried over from the previous JK BMS's, so unless you reach a Cell RCV on a regular basis the SOC drifts away from reality. This is even after calibrating the V and A on the JK BMS's, you can see at very low A's that the JK is hunting above and below 0A by some margin and overnight with low draws the % drift is the worst.

This you would think is application to this project as a solution but JK changed the RS485 protocol for the JK PB series so solutions for older JK's do not carry over. I have attached the new RS485 protocol, original in Chinese and a Google Translated versions but there are other RS485 protocol's selectable which may be more suitable
[BMS.RS485.Modbus.V1.0.pdf](https://github.com/user-attachments/files/15945159/BMS.RS485.Modbus.V1.0.pdf)
[Jikong.BMS.RS485.Modbus.universal.protocol.V1.0.pdf](https://github.com/user-attachments/files/15945160/Jikong.BMS.RS485.Modbus.universal.protocol.V1.0.pdf)

This image shows the other RS485 protocols in use by the JK PB

![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/32205016/bea254ed-73b2-47f3-bab6-0fa618ef8169)

On other Inverters of Chinese origin using the Pylontech Canbus they are also restricting the charge when the SOC reaches a calculated 100% despite the cell RCV not being met. The SMA SI does not have that issue, just the inaccurate SOC.

I have also attached the SMA Canbus protocol used by the JK which they call FSS (an obscure ref to the PDF document footer note)

Basic question is would these latest JK BMS be included at some point into the project.

[SMA CAN protocol.pdf](https://github.com/user-attachments/files/15945176/SMA.CAN.protocol.pdf)

**Comment by ArminJo** [comment]
Thanks for the documents!
I can include it, but I have no device to test it, so it will probably not work 😞 .
What is the purpose of that, since you now can attach the BMS directly to a Deye inverter and need no converter any more?

**Comment by kommando828** [comment]
Thanks for the reply, I can and do get full closed loop but the SOC inaccuracy builds up over time and the SMA Sunny Island reacts to low SOC's but ignores the Battery voltage in closed loop. So looking for a gatekeeper to intervene, giving a more accurate SOC but retaining the Canbus closed loop which gives better control of charging than open loop.
